File: server/src/repositories/decades.ts
import { getMovies } from './movies.js'

export async function getDecades() {
    const movies = await getMovies()
    const decades = new Set<number>()

    movies.forEach((movie) => {
        if (!movie.year) return
        const decade = movie.year - (movie.year % 10)
        decades.add(decade)
    })

    return [...decades].sort((a, b) => a - b)
}

File: api/app/repositories/decades.py
from app.db import cursor


async def get_decades() -> list[int]:
    async with cursor() as cur:
        await cur.execute(
            """
            SELECT
                DISTINCT year - year % 10 AS decade
            FROM movies
            ORDER BY decade;
            """
        )
        rows = await cur.fetchall()
    return [r[0] for r in rows if r[0] is not None]
